migrations/2022-12-05-110642_registration_mode/up.sql

```diff
@@ -0,0 +1,25 @@
+-- create enum for registration modes
+create type registration_mode_enum as enum
+    ('closed', 'require_application', 'open');
```

Comment:
Instead of making this exclusive, it should probably be an additional boolean option. For example, `require_email` isn't exclusive to `require_application`, its yet another ability / check that could be done. 

A require application, and require reviewing content, *could* both be active at the same time.

IE a boolean for `require_application`, and another for `require_review_content`.